packages/network/lib/uri.ts

```diff
@@ -10,24 +10,24 @@ import url, { URL } from 'url'
 
 // yup, protocol contains a: ':' colon
 // at the end of it (-______________-)
-const DEFAULT_PROTOCOL_PORTS = {
+const DEFAULT_PROTOCOL_PORTS: Record<string, string> = {
   'https:': '443',
   'http:': '80',
 }
 
 const DEFAULT_PORTS = _.values(DEFAULT_PROTOCOL_PORTS)
 
-const portIsDefault = (port) => {
+const portIsDefault = (port: string | null) => {
   return port && DEFAULT_PORTS.includes(port)
 }
 
-const parseClone = (urlObject) => {
+const parseClone = (urlObject: any) => {
   return url.parse(_.clone(urlObject))
 }
```

Comment:
This is really weird code. [`url.parse` expects `string` as its first arg](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost), but our code is written for the case that it's not a `string` but an object.

I'll check this behavior and fix this in the part 2 of this PR.